activerecord/lib/active_record/relation.rb

```diff
@@ -670,7 +670,18 @@ def values
     end
 
     def inspect
-      entries = records.take([limit_value, 11].compact.min).map!(&:inspect)
+      num_to_take = [limit_value, 11].compact.min
+      # A relation may be initialized with values that were not loaded from the
+      # database. Therefore, if there are values, they must be preserved by not
+      # loading from the database; else, use a new relation with a limit, to
+      # avoid loading an unbounded number of records into memory unnecessarily.
+      entries =
+        if values.any?
```

Comment:
Apparently, not. There may be situations where `values.any?` evaluates to `true` and `@records` is not initialized.

Tests FTW: When using `@records.present?` and running tests, I get this warning, `lib/active_record/relation.rb:679: warning: instance variable @records not initialized`, multiple times, and the following test fails:
```
  1) Failure:
AssociationProxyTest#test_inspect_does_not_reload_a_not_yet_loaded_target [/srv/amir/clones/rails/activerecord/test/cases/associations_test.rb:193]:
Expected /message: "new developer added"/ to match "#<ActiveRecord::Associations::CollectionProxy []>".
```

@arthurnn: Do you think it may be worth mentioning `@records` specifically in the comment above? I tried to make it clear in the comment, that there was a reason behind choosing `values.any?` in particular, but it may be worth detailing why `@records.present?` was ruled out.